src/analyses/base.ml

```diff
@@ -1224,13 +1223,13 @@ struct
     | Q.EvalLength e -> begin
         match eval_rv_address (Analyses.ask_of_ctx ctx) ctx.global ctx.local e with
         | `Address a ->
-          let slen = List.map String.length (AD.to_string a) in
+          let slen = List.of_seq @@ (Seq.map String.length)@@ List.to_seq (AD.to_string a) in
           let lenOf = function
             | TArray (_, l, _) -> (try Some (lenOfArray l) with LenOfArray -> None)
             | _ -> None
           in
-          let alen = List.filter_map (fun v -> lenOf v.vtype) (AD.to_var_may a) in
-          let d = List.fold_left ID.join (ID.bot_of (Cilfacade.ptrdiff_ikind ())) (List.map (ID.of_int (Cilfacade.ptrdiff_ikind ()) %BI.of_int) (slen @ alen)) in
+          let alen = List.of_seq @@ Seq.filter_map (fun v -> lenOf v.vtype)@@ List.to_seq (AD.to_var_may a) in
```

Comment:
Same question as above, as there is only a `List.filter_map` that is replaced. 
Maybe, if the `alen` is only used once, it would not have to be converted to a `list` here?